nimble/host/src/ble_l2cap_coc.c

```diff
@@ -29,6 +29,9 @@
 
 #define BLE_L2CAP_SDU_SIZE              2
 
+#define BLE_L2CAP_SET_FLAG(flags, flag) (flags |= (1 << flag))
+#define BLE_L2CAP_CLEAR_FLAG(flags, flag) (flags &= ~(1 << flag))
```

Comment:
you should better create `static inline`s which can set and clear bit in complete cid bitmask, it will make code easier to follow. also looking on code below an extra function to get 1st free bit in a cid bitmask will be useful.